sql/src/main/antlr/OpenSearchSQLParser.g4

```diff
@@ -304,7 +304,7 @@ functionCall
     ;
 
 highlightFunction
-    : HIGHLIGHT LR_BRACKET relevanceField RR_BRACKET
+    : HIGHLIGHT LR_BRACKET relevanceField RR_BRACKET (DOT qualifiedName)*
```

Comment:
Querying complex values is independent of highlight -- let's not tackle it as part of this PR.

SQL plugin does not support querying complex values returned by an expression and `highlight('*')` is an expression.